File: week2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: \n");
    int textLength = strlen(text);
// basic values to use later
// needed as wordCoutn counts the ' ' so it will miss the 1st word
// (1)
    int wordCount = 1;
    int sentenceCount = 0;
    int letterCount = 0;

    for (int i = 0; i < textLength; i++){
        text[i] = tolower(text[i]);
        if (text[i] >= 'a' && text[i] <= 'z') // as everything is lowercase all letters have to be between a and z.
        {
            letterCount ++;
        }
        else if(text[i] == ' ') // assumes all words are seperated by spaces
        {
            wordCount ++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentenceCount ++;
        }
    }
    float index = 0.0588 * ((float) letterCount / (float) wordCount  * 100) - 0.296 * ((float) sentenceCount /
                (float) wordCount  * 100) - 15.8;
    index = round(index);
    if (index >= 0 && index < 16)
    {
        printf("Grade %i\n", (int) round(index));
    }
    else if (index < 0)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }

}

// (1) https://www.reddit.com/r/cs50/comments/ez9bpx/readability/

File: week6/readability.py
import re

from cs50 import get_string

def main():
    text = get_string("Please input text:\n")
    readability(text)

def readability(text):
    letter_count = 0.0
    word_count = 1.0
    sentence_count = 0.0
    for i in text:
        # next line from https://www.tutorialgateway.org/python-program-to-check-character-is-alphabet-or-not/
        if i.isalpha():
            letter_count += 1
        elif i.isspace():
            word_count += 1
        # elif i == ('.' or '!' or '?'):
    sentence_count += len(re.findall(r'\.!?', text))# += 1
    print(f"letters: {letter_count}")
    print(f"words {word_count}")
    print(f"sentences {sentence_count}")

    index = 0.0588 * ( letter_count / word_count * 100) - 0.296 * (sentence_count /
            word_count * 100) - 15.8
    print("Index:")
    print(index)

    if index < 1:
        print("Before Grade 1")
    elif (index > 16):
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")

main()